test/module/shared_model/builders/protobuf/transaction_responses/proto_transaction_response_builder.cpp

```diff
@@ -21,156 +21,63 @@
-}
+TYPED_TEST(ProtoTransactionStatusBuilderTest, TestStatusType) {
+  using StatusType = typename TypeParam::IfaceType;
 
-/**
- * @given expected transaction stateful valid status and hash
- * @when model object is built using these status and hash
- * @then built object has expected status and hash
- */
-TEST(ProtoTransactionStatusBuilderTest, TestStatefulValidStatus) {
-  using StatefulValidStatusType =
-      shared_model::interface::StatefulValidTxResponse;
-
-  auto expected_status = iroha::protocol::STATEFUL_VALIDATION_SUCCESS;
-  auto expected_hash = std::string(32, '1');
-
-  auto stateful_invalid_response =
-      TransactionStatusBuilder()
-          .statefulValidationSuccess()
-          .txHash(shared_model::crypto::Hash(expected_hash))
-          .build();
-
-  boost::apply_visitor(verifyType<StatefulValidStatusType>(),
-                       stateful_invalid_response.get());
-
-  auto proto_status = stateful_invalid_response.getTransport();
-  ASSERT_EQ(proto_status.tx_status(), expected_status);
-  ASSERT_EQ(proto_status.tx_hash(), expected_hash);
-}
-
-/**
- * @given expected transaction commit status and hash
- * @when model object is built using these status and hash
- * @then built object has expected status and hash
- */
-TEST(ProtoTransactionStatusBuilderTest, TestCommittedStatus) {
-  using CommittedStatusType = shared_model::interface::CommittedTxResponse;
-
-  auto expected_status = iroha::protocol::COMMITTED;
-  auto expected_hash = std::string(32, '1');
-
-  auto committed_response =
-      TransactionStatusBuilder()
-          .committed()
-          .txHash(shared_model::crypto::Hash(expected_hash))
-          .build();
-
-  boost::apply_visitor(verifyType<CommittedStatusType>(),
-                       committed_response.get());
-
-  auto proto_status = committed_response.getTransport();
-  ASSERT_EQ(proto_status.tx_status(), expected_status);
-  ASSERT_EQ(proto_status.tx_hash(), expected_hash);
-}
-
-/**
- * @given expected transaction not received status and hash
- * @when model object is built using these status and hash
- * @then built object has expected status and hash
- */
-TEST(ProtoTransactionStatusBuilderTest, TestNotReceivedStatus) {
-  using NotReceivedStatusType = shared_model::interface::NotReceivedTxResponse;
-
-  auto expected_status = iroha::protocol::NOT_RECEIVED;
+  auto expected_status = TypeParam::status;
   auto expected_hash = std::string(32, '1');
 
-  auto not_received_response =
-      TransactionStatusBuilder()
-          .notReceived()
-          .txHash(shared_model::crypto::Hash(expected_hash))
-          .build();
+  auto response = (TransactionStatusBuilder().*TypeParam::member)()
+                      .txHash(shared_model::crypto::Hash(expected_hash))
+                      .build();
 
-  boost::apply_visitor(verifyType<NotReceivedStatusType>(),
-                       not_received_response.get());
+  boost::apply_visitor(verifyType<StatusType>(), response.get());
```

Comment:
Now we have specified visitor. We can replace `verifyType` function with it and also remove `transaction_builders_common.hpp`